lib/toSyllabicSymbols.js

```diff
@@ -0,0 +1,144 @@
+
+// Other symbols/signs have no weight, therfore remove before processing
+const zeroUnitCharacters = [
+  '੍ਰ',
+  'ੵ',
+  'ੑ',
+  '੍ਵ',
+  '੍ਹ',
+  '੍ਟ',
+  '੍ਨ',
+  '੍ਯ',
+  '੍ਚ',
+  '੍ਤ',
+  '੍',
+  'ਿ',
+  'ੁ',
+  'ਂ',
+  'ਃ',
+  '।',
+  '॥',
+  '☬',
+  '਼',
+  '❁',
+]
+
+// For the rest, need to analyze the string
+// Any base character on it's own counts as one syllable
+// Adding any number of long/deeragh sounds will count as two syllables
+const syllableIdentifier = {
+  baseCharacter: 'b',
+  deeraghModifier: 'a',
+}
+
+// The list of base characters for analysis
+const baseCharacters = [
+  'ਙ',
+  'ੳ',
+  'ਅ',
+  'ਬ',
+  'ਭ',
+  'ਚ',
+  'ਛ',
+  'ਦ',
+  'ਧ',
+  'ੲ',
+  'ਡ',
+  'ਢ',
+  'ਗ',
+  'ਘ',
+  'ਹ',
+  'ਜ',
+  'ਝ',
+  'ਕ',
+  'ਖ',
+  'ਲ',
+  'ਲ਼',
+  'ਮ',
+  'ਨ',
+  'ਪ',
+  'ਫ',
+  'ਤ',
+  'ਥ',
+  'ਰ',
+  'ਸ',
+  'ਸ਼',
+  'ਟ',
+  'ਠ',
+  'ਵ',
+  'ੜ',
+  'ਣ',
+  'ਯ',
+  'ਜ਼',
+  'ਗ਼',
+  'ਖ਼',
+  'ਫ਼',
+  'ਞ',
+]
+
+// The list of long sound characters for analysis
+const deeraghModifiers = [
+  'ੰ',
+  '੍ਹੂ',
+  'ੀ',
+  'ੂ',
+  'ੇ',
+  'ੈ',
+  'ੋ',
+  'ੌ',
+  'ਾਂ',
+  'ਾ',
+  'ੱ',
+]
+
+// Map one-unit syllables to the string '1' and two-unit syllables to the string '2'
+// Otherwise map each character to a syllableIdentifier, for further processing/analysis
+const syllabicMapping = Object.entries( {
+  1: [ ...oneUnitSyllables ],
+  2: [ ...twoUnitSyllables ],
+  [ syllableIdentifier.baseCharacter ]: [ ...baseCharacters ],
+  [ syllableIdentifier.deeraghModifier ]: [ ...deeraghModifiers ],
+} ).reduce( ( newObject, [ weight, characters ] ) => {
+  const characterMap = characters.reduce( ( characterMapObject, [ character ] ) => ( {
+    ...characterMapObject,
+    [ character ]: weight,
+  } ), {} )
+  return ( {
+    ...newObject,
+    ...characterMap,
+  } )
+}, {} )
+
+// Add any missing rules which do not fit in the above patterns
+syllabicMapping[ 'ੴ' ] = '21 2221' // Ik Oankaar / ਇੱਕ ਓਅੰਕਾਰ
+syllabicMapping[ ' ' ] = ' ' // Preserve spacing between words
+
+// Create the Regex rule that will be used to strip characters irrelevant to syllable weighting
+const regexToStrip = new RegExp( getRegexGroup( zeroUnitCharacters ), 'g' )
+
+/**
+ * Represents text in syllables according to Sanskrit prosody, Pingala, Matra/Meter/Morae
+ * @param {String} text The string to convert
+ * @return {String} A syllabic representation of 1's (laghu/light/short) and 2's (guru/heavy/long).
+ * @example
+ * toSyllabicSymbols( 'ਪ੍ਰਭੂ ਪ੍ਰੇਮੀ ਪੜ੍ਹ ਚੜ੍ਹ ਦ੍ਵੈਤ' )
+ * // expected output: '12 22 11 11 21'
+ */
+const toSyllabicSymbols = ( text ) => toUnicode( text ) // Sanitize input
+  .replace( regexToStrip, '' ) // Remove zero weight characters
+  .split( '' ) // Break apart all characters for mapping
+  .map( ( value ) => syllabicMapping[ value ] ) // Map according to rules above ('1, 2, b, a')
+  .join( '' ) // Combine to check sequences
+  .replace( /[a]+/g, ( v ) => v.slice( 0, 1 ) ) // Only one long sound is enough to modify base char
```

Comment:
Why use a character set here for a single character?

Also, it may be more explicit to just reference `a` in the replacement value. I ran some console examples to confirm your code was doing what I thought, which is the reason for this suggestion.

```suggestion
  .replace( stripMultipleDeeraghRegex, symbolIdentifiers.deeraghModifier ) // Only one long sound is enough to modify base char
```

with `const stripMultipleDeeraghRegex = new RegExp( `${symbolIdentifiers.oneUnit}+`, 'g')` somewhere above.

I guess my general point here is about removing floating strings and regexes if possible.

A very minor performance point here too, pre-calculating a regex with `new RegExp` is faster for repeated usage than an inline `/ba/` regex. I think this reason and the above `const stripMultipleDeeraghRegex` is what led me down the post-replacements route, since it became tedious for me declare all the variables, but perhaps you can create a Regexes object above, so you can reference it as 

` .replace( regexes.stripMultipleDeeragh, symbolIdentifiers.deeraghModifier )`, which reads more like an English sentence.